src/library/export/dlglibraryexport.cpp

```diff
@@ -0,0 +1,286 @@
+            [this]() { m_pCratesList->setEnabled(true); });
+
+    // Button to allow ability to browse for the export directory.
+    auto pExportDirBrowseButton = make_parented<QPushButton>(tr("Browse"));
+    connect(pExportDirBrowseButton,
+            &QPushButton::clicked,
+            this,
+            &DlgLibraryExport::browseExportDirectory);
+    auto pExportDirLayout = make_parented<QHBoxLayout>();
+    pExportDirLayout->addWidget(m_pExportDirectoryTextField);
+    pExportDirLayout->addWidget(pExportDirBrowseButton);
+
+    auto pFormLayout = make_parented<QFormLayout>();
+    pFormLayout->addRow(tr("Export directory"), pExportDirLayout);
+    pFormLayout->addRow(tr("Database version"), m_pVersionCombo);
+    pFormLayout->addRow(m_pExistingDatabaseLabel);
+
+    // Buttons to begin the export or cancel.
+    auto pExportButton = make_parented<QPushButton>(tr("Export"));
+    pExportButton->setDefault(true);
+    connect(pExportButton, &QPushButton::clicked, this, &DlgLibraryExport::exportRequested);
+    auto pCancelButton = make_parented<QPushButton>(tr("Cancel"));
+    connect(pCancelButton, &QPushButton::clicked, this, &QDialog::reject);
+
+    // Arrange action buttons at bottom of dialog.
+    auto pButtonBarLayout = make_parented<QHBoxLayout>();
+    pButtonBarLayout->addStretch(1);
+    pButtonBarLayout->addWidget(pExportButton);
+    pButtonBarLayout->addWidget(pCancelButton);
+
+    auto pLayout = make_parented<QGridLayout>();
+    pLayout->setColumnStretch(0, 1);
+    pLayout->setColumnStretch(1, 2);
+    pLayout->addWidget(m_pWholeLibraryRadio, 0, 0);
+    pLayout->addWidget(m_pCratesRadio, 1, 0);
+    pLayout->addWidget(m_pCratesList, 2, 0);
+    pLayout->addLayout(pFormLayout, 0, 1, 3, 1);
+    pLayout->addLayout(pButtonBarLayout, 3, 0, 1, 2);
+
+    setLayout(pLayout);
+    setWindowTitle(tr("Export Library to Engine Prime"));
+
+    show();
+    raise();
+    activateWindow();
+}
+
+void DlgLibraryExport::refresh() {
+    // Refresh the list of crates.
+    populateCrates(m_pCratesList, *m_pTrackCollectionManager->internalCollection());
+
+    // Check whether a database already exists in the specified directory.
+    checkExistingDatabase();
+}
+
+void DlgLibraryExport::setSelectedCrate(std::optional<CrateId> crateId) {
+    if (!crateId) {
+        m_pWholeLibraryRadio->setChecked(true);
+        m_pCratesList->setEnabled(false);
+        return;
+    }
+
+    m_pCratesRadio->setChecked(true);
+    m_pCratesList->setEnabled(true);
+    for (auto i = 0; i < m_pCratesList->count(); ++i) {
+        auto* pItem = m_pCratesList->item(i);
+        auto currCrateId = pItem->data(Qt::UserRole).toInt();
+        if (currCrateId == crateId->value()) {
+            m_pCratesList->setCurrentItem(pItem);
+            return;
+        }
+    }
+}
+
+void DlgLibraryExport::browseExportDirectory() {
+    QString lastExportDirectory =
+            m_pConfig->getValue(ConfigKey("[Library]", "LastLibraryExportDirectory"),
```

Comment:
Define constants instead of duplicating string literals

Reply:
Done.